test/suite/test_rollback_to_stable15.py

```diff
@@ -0,0 +1,133 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+import wiredtiger, wttest, unittest
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_rollback_to_stable15.py
+# Test that roll back to stable handles updates present in the
+# update-list for both fixed length and variable length column store.
+# Eviction is set to false, so that everything persists in memory.
+class test_rollback_to_stable15(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=200MB,statistics=(all),debug_mode=(eviction=false)'
+    session_config = 'isolation=snapshot'
+    key_format_values = [
+        ('column', dict(key_format='r')),
+        ('integer', dict(key_format='i')),
+    ]
+    value_format_values = [
+        # Fixed length
+        ('fixed', dict(value_format='8t')),
+        # Variable length
+        ('variable', dict(value_format='i')),
+    ]
+    scenarios = make_scenarios(key_format_values, value_format_values)
+
+    def check(self, check_value, uri, nrows, read_ts):
+        session = self.session
+        if read_ts == 0:
+            session.begin_transaction()
+        else:
+            session.begin_transaction('read_timestamp=' + timestamp_str(read_ts))
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def test_rollback_to_stable(self):
+        # Create a table.
+        uri = "table:tmp"
+        nrows = 2000
+        create_params = 'key_format={},value_format={}'.format(self.key_format, self.value_format)
+        self.session.create(uri, create_params)
+        cursor =  self.session.open_cursor(uri)
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        value20 = 0x20
+        value30 = 0x30
+        value30 = 0x40
+        value40 = 0x50
+
+        #Insert value20 at timestamp 2
+        for i in range(1, nrows): # make a variable.
+            self.session.begin_transaction()
+            cursor[i] = value20
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(2))
+
+        #First Update to value 30 at timestamp 5
+        for i in range(1, nrows):
+            self.session.begin_transaction()
+            cursor[i] = value30
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(5))
+
+        #Set stable timestamp to 2
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(2))
+        self.conn.rollback_to_stable()
+        # Check that only value20 is available
+        self.check(value20, uri, nrows - 1, 2)
```

Comment:
Instead of passing `nrows -1`, can you change the range from `0, nrows` in the upper for loops.

Reply:
If we can it to `0` until `nrows - 1`, whenever we set a key it needs to be `i+1` as keys cannot be 0 in column store, is that alright?

Reply:
Thanks. I forgot that 0 is used as recno. You can retain the original code change.